src/main/java/com/google/api/gax/retrying/RetryFutureImpl.java

```diff
@@ -74,22 +74,28 @@
 
   @Override
   protected boolean set(ResponseT value) {
-    return super.set(value);
+    synchronized (lock) {
+      return super.set(value);
+    }
   }
 
   @Override
   protected boolean setException(Throwable throwable) {
-    return super.setException(throwable);
+    synchronized (lock) {
+      return super.setException(throwable);
+    }
   }
 
   @Override
   public boolean cancel(boolean mayInterruptIfRunning) {
     synchronized (lock) {
-      boolean rv = super.cancel(mayInterruptIfRunning);
       if (callbackFutureCallback != null) {
-        callbackFutureCallback.attemptFuture.cancel(mayInterruptIfRunning);
+        boolean rv = callbackFutureCallback.attemptFuture.cancel(mayInterruptIfRunning);
+        super.cancel(mayInterruptIfRunning);
+        return rv;
```

Comment:
Changed, though I will miss rvs. The idea behind using generic "returned value" as a name for returned value is that it is usually difficult to come up with a reasonable name for that variable anyways, and what exactly the value is about is always explained by the name of the method itself.